treeherder/config/settings.py

```diff
@@ -18,20 +20,25 @@
 
 env = environ.Env()
 
+# Checking for OS type
+IS_WINDOWS = "windows" in platform.system().lower()
+
 # Top Level configuration
 DEBUG = env.bool("TREEHERDER_DEBUG", default=False)
```

Comment:
With the new check we need to change the original assumptions.

If `HEROKU_APP_NAME` is not defined then let's assume we're running on `DEBUG` mode.
The only place we want to Treeherder to run as a production system is in Heroku.
If we ever need to move to a different provider we can deal with it then.

Do this instead:
`DEBUG = not env.bool("HEROKU_APP_NAME", default=False)`

Reply:
@shubhank-saxena  @armenzg please do not do this:  Have one variable per concept, and do not overload them: There may be a time we want to consume-and-test the `HEROKU_APP_NAME`.  For example, the `extract` code could use it instead of `NEW_RELIC_APP_NAME`, which is a bit uglier.

Reply:
I'm curious, do you have an instance where that variable (or similar) will be consumed in a context different than while running as a Heroku context?

I want the code to clearly fail when part of a Heroku set up. For instance, when creating a new app. Perhaps, if `HEROKU_APP_NAME` is defined fail saying "You're in a Heroku context please set up the env variable to `PRODUCTION_SETUP=True`." before checking for all other required env variables.

What do you say?

Reply:
Haven't made the changes in this yet!

Reply:
> I'm curious, do you have an instance where that variable (or similar) will be consumed in a context different than while running as a Heroku context?

Yes, I have a specific test to ensure `NEW_RELIC_APP_NAME` is properly used when pushing to BigQuery.  I could have used `HEROKU_APP_NAME`, if I knew it existed at the time because it is has nicer values.

When linking DEBUG and HEROKU_APP_NAME, you are complicating the interaction:  If I set HEROKU_APP_NAME, then I will find Treeherder will behave differently; it will take time to find the link.  

If you must link the two, I have no problem with throwing errors when invalid combinations happen; like you suggest 

>  You're in a Heroku context please set up the env variable to PRODUCTION_SETUP=True.

Adding the link is more complicated, but at least it is discoverable.